plugins/Diagnostics/Diagnostic/ForceSSLCheck.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+use Piwik\Config;
+use Piwik\Translation\Translator;
+
+/**
+ * Check that Matomo is configured to force SSL.
+ */
+class ForceSSLCheck implements Diagnostic
+{
+    /**
+     * @var Translator
+     */
+    private $translator;
+
+    public function __construct(Translator $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function execute()
+    {
+        $label = $this->translator->translate('General_ForcedSSL');
+
+        $forceSSLEnabled = (Config::getInstance()->General['force_ssl'] == 1);
+
+        if ($forceSSLEnabled) {
+            return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_OK));
+        }
+
+        $comment = $this->translator->translate('General_ForceSSLRecommended');
+
+        return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_WARNING, $comment));
```

Comment:
Also, should this check run if matomo isn't currently on `https`? If it isn't and a user adds `force_ssl = 1`, they may be confused why it doesn't work.

Reply:
It should be checked and warned about! But with another text indicating the fact that a certificate need to be deployed first (with a link to eff's org maybe?).